doc/adr/release.md

```diff
@@ -0,0 +1,1019 @@
+`release.yaml` file at the root of the git repository using IOGX.
+
+The release configuation file will have the fallowing options...
+
+##### `title.prefix`
+
+- Type: String
+- Default: Empty String
+
+A prefix combined with the [Release Identifier](#release-identifier)
+to form a release title to be used in content where human readability is important.
+
+##### `description.text`
+
+- Type: String
+- Default: Empty String
+
+A description of the release.
+
+##### `description.include-github-generated-release-notes`
+
+- Type: Bool
+- Default: False
+
+Whether to include the release notes generated by GitHub in the descriptions of
+the release.
+
+##### `git.tag.always-publish`
+
+- Type: Bool
+- Default: False
+
+If the git tag should be published even if no release objectives call for it.
+
+##### `git.tag.prefix`
+
+- Type: String
+- Default: Empty String
+
+A string prepended to the [Release Identifier](#release-identifier) to
+form the git tag representing the release in git repositories.
+
+##### `github.release.always-publish`
+
+- Type: Bool
+- Default: False
+
+If the GitHub release should be published even if no other release objectives
+call for it.
+
+##### `github.release.assets`
+
+- Type: Mapping (String -> `flake-output-path`)
+- Default: Empty Map
+
+A mapping from the desired name of the asset in the GitHub release to the file
+to exist under that asset name.
+
+> Note: All values (`flake-output-path`s) must resolve to files.
+
+<!-- -->
+> Note: There cannot be any naming conflicts between names to be used as asset
+names in the GitHub release.
+
+Example
+
+The configuration...
+
+```yaml
+github:
+  release:
+    assets:
+      foo-x86_64-linux:
+        output: packages.x86_64-linux.foo
+        path: /lib/foo.so
+      foo2-x86_64-linux:
+        output: packages.x86_64-linux.foo
+        path: /lib/foo2.so
+      bar:
+        output: packages.x86_64-linux.bar
+      biz-x86_64-linux: biz
+```
+
+results in the files at these paths being uploaded to GitHub.
+
+```console
+/nix/store/XXXXXXXX-foo-derivation/lib/foo.so -> foo-x86_64-linux
+/nix/store/XXXXXXXX-foo-derivation/lib/foo2.so -> foo2-x86_64-linux
+/nix/store/XXXXXXXX-bar-derivation -> bar
+/nix/store/XXXXXXXX-biz-derivation -> biz-x86_64-linux
+```
+
+###### `flake-output-path` Type
+
+In the release configuration file, a `flake-output-path` is a value that
+represents a node from a path in a derivation accessible from a project's flake
+`outputs`. To use the node, the derivation needs to be built first.
+
+A `flake-output-path` value can take one of the forms...
+
+1. A string representing a selector of a derivation from the flake's `outputs`
+2. A map containing the keys
+    - `output` - A string representing a selector of a derivation from the flake's
+    `outputs`.
+    - `path` - An optional file path that will be used to select a file from the
+    derivation once it has been built.
+
+Examples
+
+```yaml
```

Comment:
These examples don't seem to follow the expected format?

Reply:
That is just meant to be a list of `flake-output-path` values. Each value should fit the specified format of a `flake-output-path`. Maybe the example makes things more confusing.